sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/RxDocumentClientImpl.java

```diff
@@ -4727,14 +4727,25 @@ private <T> Flux<FeedResponse<T>> readFeed(
         ResourceType resourceType,
         Class<T> klass,
         String resourceLink) {
+        DocumentClientRetryPolicy retryPolicy = this.resetSessionTokenRetryPolicy.getRequestPolicy(null);
+        return ObservableHelper.fluxInlineIfPossibleAsObs(
+            () -> readFeedInternal(options, resourceType, klass, resourceLink, retryPolicy),
+            retryPolicy);
+    }
+
+    private <T> Flux<FeedResponse<T>> readFeedInternal(
```

Comment:
NIT: IMO it could help to avoid misunderstandings to call this method something like nonDocumentReadFeedInternal instead - we assert that it is not accidentally used for document operations - but from a code reading perspective I guess it would make it even more clear.

Reply:
yea good idea, updated